src/redux/reducers.ts

```diff
@@ -0,0 +1,51 @@
+import { Action, Reducer } from "redux";
+import { CanvasState } from "./state";
+import { Figure } from "../models/figure";
+import * as Actions from "./actions";
+
+export function figures(): Reducer<CanvasState> {
+  const initialState = {
+    figures: new Array<Figure>()
+  };
+
+  const figuresExcept = (figures: Figure[], index: number): Figure[] => {
+    return figures.slice(0, index).concat(figures.slice(index + 1));
+  };
+
+  return (state: CanvasState = initialState, action: Action) => {
+    switch(action.type) {
+      case Actions.ADD_FIGURE:
+        return {
+          figures: [ ...state.figures, (action as Actions.AddFigureAction).figure ]
+        }
+
+      case Actions.REMOVE_FIGURE:
+        var index = (action as Actions.RemoveFigureAction).index;
+        var figures = state.figures;
```

Comment:
it's recommended to use `let` or `const` instead of `var`. these will be scoped to the block